auto_sizing/targets.py

```diff
@@ -1,36 +1,33 @@
-from typing import List, Dict, Tuple, Optional
-from pathlib import Path
-import toml
 from datetime import datetime
+from typing import Dict, List, Optional, TextIO
+
 import attr
-from mozanalysis.segments import Segment, SegmentDataSource
-from mozanalysis.metrics import Metric, DataSource
+import toml
 from mozanalysis.config import ConfigLoader
+from mozanalysis.metrics import DataSource, Metric
+from mozanalysis.segments import Segment, SegmentDataSource
 from mozanalysis.utils import add_days
 
-from .utils import dict_combinations, default_dates_dict
-from .errors import (
-    MetricsTagNotFoundException,
-    SegmentsTagNotFoundException,
-    SegmentDataSourcesTagNotFoundException,
-    DataSourcesTagNotFoundException,
-)
+from .errors import MetricsTagNotFoundException, SegmentsTagNotFoundException
+from .utils import default_dates_dict, dict_combinations
 
 
 class SegmentsList:
     """Builds list of Segments from list of dictionaries"""
 
-    def from_repo(
-        self, target_list: Dict, app_id: str, start_date: Optional[str] = None
-    ) -> List[Segment]:
+    def from_repo(self, target_list: Dict, app_id: str, start_date: str = "") -> List[Segment]:
```

Comment:
Just an idea, but might be more helpful to define the `app_id`'s type with the allowed values using an enum or a `Literal`. I'm not sure what the most pythonic method would be though, and what you have is certainly valid.

```suggestion
    def from_repo(self, target_list: Dict, app_id: Literal["firefox_desktop", "firefox_ios", "fenix"], start_date: str = "") -> List[Segment]:
```